tesseract_core/runtime/experimental.py

```diff
@@ -248,11 +248,93 @@ def require_file(file_path: PathLike) -> Path:
     return file_path
 
 
+class TesseractReference:
+    """Allows passing a reference to another Tesseract as input."""
+
+    def __init__(self, tesseract: Any) -> None:
+        self._tesseract = tesseract
+
+    def __getattr__(self, name: str) -> Any:
+        """Delegate attribute access to the underlying Tesseract instance."""
+        return getattr(self._tesseract, name)
+
+    @classmethod
+    def _get_tesseract_class(cls) -> type:
+        """Lazy import of Tesseract class. Avoids hard dependency of Tesseract runtime on Tesseract SDK."""
+        try:
+            from tesseract_core import Tesseract
+
+            return Tesseract
+        except ImportError:
+            raise ImportError(
+                "Tesseract class not found. Ensure tesseract_core is installed and configured correctly. "
+                "When building a Tesseract, you can add tesseract_core to your tesseract_requirements.txt."
```

Comment:
Let's avoid this blanket suggestion, or alternatively specify it's not just for "building" but more like something you can do while you are "developing" stuff. I'd avoid anyway and document properly how to do this HOT stuff once it's more tried and tested
```suggestion
```